src/praisonai/praisonai/bots/whatsapp.py

```
@@ -835,6 +862,57 @@ async def send_message(
+        """Download WhatsApp media by id and cache a validated local path.
+
+        Returns the cached file path, or ``None`` when media handling is
+        disabled, the id is missing, or download/validation fails. The agent's
+        vision capability acts on the returned path (Issue #2350).
+        """
+        max_bytes = getattr(self.config, "max_inbound_media_bytes", 0)
+        if not media_id or not max_bytes or max_bytes <= 0:
+            return None
+        try:
+            data = await self._download_media(media_id)
+        except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)